src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -2351,6 +2352,22 @@ RunDistributedExecution(DistributedExecution *execution)
 				execution->waitFlagsChanged = false;
 			}
 
+			WorkerSession *session = NULL;
+			foreach_ptr(session, execution->sessionList)
+			{
```

Comment:
I think we only need to do this after RebuildWaitEventSet or RebuildWaitEventSetFlags, so maybe move into a separate function and call it after those calls.